src/tactic/induction.lean

```diff
@@ -1213,7 +1213,12 @@ remember the value of the major premise.
 meta def eliminate_expr (generate_induction_hyps : bool) (major_premise : expr)
   (eq_name : option name := none) (gm := generalization_mode.generalize_all_except [])
   (with_names : list name := []) : tactic unit := do
-  num_reverted ← revert_reverse_dependencies_of_hyp major_premise,
```

Comment:
Is there a reason that you didn't fix `revert_reverse_dependencies_of_hyp` instead?

Reply:
The whole `dependencies_of_hyp` family works only for local constants. (Hence the issue when `major_premise` is not a local constant.) Generalising `dependencies_of_hyp` to `dependencies_of` would be nice, but also a bit of work.

Actually now that I think about it, maybe not so much work. I'll look into it.